File: packages/audio/android/src/main/java/dev/tcode/thinmpr/audio/service/contract/AlbumServiceContract.kt
package dev.tcode.thinmpr.audio.service.contract

interface AlbumServiceContract {
    fun getAllAlbums(): List<Map<String, Any>> 

    fun getAlbumById(id: String): Map<String, Any>?

    fun getAlbumsByArtistId(artistId: String): List<Map<String, Any>>

    fun getRecentAlbums(count: Int): List<Map<String, Any>>
}

File: packages/audio/ios/Service/Contract/AlbumServiceContract.swift
protocol AlbumServiceContract {
    func getAllAlbums() -> [[String: Any]]

    func getAlbumById(albumId: String) -> [String: Any]?

    func getAlbumsByArtistId(artistId: String) -> [[String: Any]]

    func getRecentAlbums(count: Int) -> [[String: Any]]
}
